REST_based_Auth_API/my_rest_api/core/views.py

```diff
@@ -0,0 +1,371 @@
+    serializer_class = MyTokenObtainPairSerializer
+
+
+class HelloView(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        content = {"message": "Hello Faraz!"}
+        return Response(content)
+
+
+class RegisterUserView(generics.GenericAPIView):
+    serializer_class = UserSerializer
+    permission_classes = (AllowAny,)
+    queryset = User.objects.all()
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.serializer_class(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+        user = serializer.save()
+        return Response(self.serializer_class(user).data)
+
+
+class GetUsersView(viewsets.ModelViewSet):
+    serializer_class = UserSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = User.objects.all()
+
+
+class PostView(viewsets.ModelViewSet):
+    serializer_class = PostSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = Post.objects.all()
+
+    def get_queryset(self):
+        user = self.request.user
+        return self.queryset.filter(user=user)
+
+
+class CommentView(viewsets.ModelViewSet):
+    serializer_class = CommentSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = Comment.objects.all()
+
+    def get_queryset(self):
+        post_id = self.request.data["post"]
+        comment_id = self.request.data["comment"]
+
+        if comment_id:  # checking it because it can be null
+            return self.queryset.filter(comment=comment_id)
+        return self.queryset.filter(post=post_id)
+
+    def create(self, request, *args, **kwargs):
+        signed_in_user = self.request.user.id
+        post_id = self.request.data["post"]
+        comment_id = self.request.data["comment"]
+
+        commentor = User.objects.get(pk=signed_in_user)
+        parent_post = Post.objects.get(pk=post_id)
+        if not parent_post:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        post_author = parent_post.user_id
+
+        if comment_id:
+            parent_comment = Comment.objects.get(pk=comment_id)
+            if not parent_comment:
+                return Response(status=status.HTTP_400_BAD_REQUEST)
+            comment_type = Comment.CommentType.COMMENT
+            notification_text = (
+                f"{commentor.username} has replied to a Comment on your post"
+            )
+
+        else:
+            comment_type = Comment.CommentType.POST
+            notification_text = f"{commentor.username} has Commented on your post"
+
+        serializer = self.serializer_class(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+
+        create_notification(
+            request,
+            to_user=User(post_author),
+            notification=notification_text,
+            notification_source_type=comment_type,
+            user_by=signed_in_user,
+            notification_source_id=notification_source_id,
```

Comment:
`notification_source_id` is undefined. Did you test this program before submitting PR?